### PR TITLE
fix(github): Shrink GraphQL pagination for `ExternalHostError`

### PR DESCRIPTION
**Pull request by zharinov**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/github/graphql/datasource-helper.ts [hunk 1]
```diff
@@ -30,13 +30,11 @@ function isUnknownGraphqlError(err: Error): boolean {
...
 
-function isTimeoutError(err: Error): err is TimeoutError {
-  return err instanceof TimeoutError;
-}
-
 function canBeSolvedByShrinking(err: Error): boolean {
   const errors: Error[] = err instanceof AggregateError ? [...err] : [err];
-  return errors.some((e) => isTimeoutError(e) || isUnknownGraphqlError(e));
+  return errors.some(
+    (e) => err instanceof ExternalHostError || isUnknownGraphqlError(e)
```
[the remark this answers is not in the document]
Comment (viceice) [reply lib/util/github/graphql/datasource-helper.ts:36]: isn't the timeout error nested on externalhosterror? so we can tight this some more?

[the remark this answers is not in the document]
Comment (viceice) [reply lib/util/github/graphql/datasource-helper.ts:36]: yes, https://github.com/renovatebot/renovate/blob/fb09a30b5e03e1e1211f2f7e3dc9cbc3cb8cb337/lib/types/errors/external-host-error.ts#L6

[the remark this answers is not in the document]
Comment (zharinov) [reply lib/util/github/graphql/datasource-helper.ts:36]: Yes, but other graphql we shrink for the whole class of errors: https://github.com/renovatebot/renovate/blob/main/lib/util/http/github.ts#L399

[the remark this answers is not in the document]
Comment (viceice) [reply lib/util/github/graphql/datasource-helper.ts:36]: ok

**Review by viceice — APPROVED** [review]

**Pull request merged**